json/json_test.go

```diff
@@ -1591,6 +1591,28 @@ func TestGithubIssue44(t *testing.T) {
 	}
 }
 
+type issue107Foo struct {
+	Bar *issue107Bar
+}
+
+type issue107Bar struct {
```

Comment:
Could move those struct type defs into `TestGithubIssue107` so that they don't need to be prefixed.

Reply:
It seems we can't, in the function scope the reference to the not-yet-declared type is invalid.